Queuesim/TrafficSource.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+
+from TrafficUnit import TrafficUnit
+
+class TrafficSourceBase(ABC):
+    def __init__(self, env, server):
+        self.__traffic = []
+        self.__env = env
```

Comment:
If the server already contains env then you should not keep a separate copy in this class. What will happen if the user passes in a different copy of env than contained in the server?